2-mandatory/2-water-bottle.js

```diff
@@ -15,22 +15,34 @@ You have to implement the missing features according to the specification.
 */
 
 // Here is your starting point:
+unit = 10;
+bottleIsFull = 100;
 let bottle = {
   volume: 0,
   fillUp: function () {
     // calling this function should completely fill your bottle (volume = 100);
+    return this.volume = bottleIsFull;
   },
   pour: function () {
     // calling this function should increase your bottle volume by 10 units;
+    if (this.volume !== bottleIsFull){
+      return this.volume += unit;
+    }
   },
   drink: function () {
     // calling this function should decrease your bottle volume by 10 units;
+    if (this.volume !== 0){
+      return this.volume -= unit;
+    }
+    //
   },
   isFull: function () {
     // this function should return true if your bottle is full;
+    return this.volume === bottleIsFull;
   },
   isEmpty: function () {
     // this function should return true if your bottle is empty;
+    return !this.volume;
```

Comment:
A typical check for isEmpty is to compare to 0, similar to isFull. This code does pass the tests however if volume is undefined than isEmpty will return true.